# Importing the libraries and the dataset

In [1]:
import numpy as np 
import tensorflow as tf 
import datetime 
from tensorflow.keras.datasets import fashion_mnist

# Data Preprocessing

### Loading the dataset

In [2]:
(X_train, y_train), (X_test, y_test) = fashion_mnist.load_data()

In [3]:
X_train.shape, X_test.shape

((60000, 28, 28), (10000, 28, 28))

### Normalizing the images

Normalizing the images
We divide each pixel of the image in the training and test sets by the maximum number of pixels (255).

In this way each pixel will be in the range [0, 1]. By normalizing images we make sure that our model (ANN) trains faster.

In [4]:
X_train = X_train/255.0
X_test = X_test/255.0

### Reshaping the dataset

Since we are building a fully connected network, we reshape the training set and the test set to be into the vector format

Since each image's dimension is 28x28,


we reshape the full dataset to [-1 (all elements), height * width]


In [5]:
X_train = X_train.reshape(-1,28*28)
X_test = X_test.reshape(-1,28*28)
X_train.shape , X_test.shape

((60000, 784), (10000, 784))

# Building an Artificial Neural Network

### Defining the model

Simply define an object of the Sequential model.

In [6]:
model = tf.keras.models.Sequential()

Adding a first fully-connected hidden layer

Layer hyper-parameters:

number of units/neurons: 128

activation function: ReLU

input_shape: (784, )